### ISSUE TITLE
Compare branches within neovim

### ISSUE THREAD

**Issue by farbodsz** [farbodsz/dotfiles#8]
Being able to compare any two branches within GitHub/BitBucket/etc is pretty nice.

We can kinda do this with [`vim-fugitive`](https://github.com/tpope/vim-fugitive/), which displays diffs from multiple files in separate tabs. But opening these up and navigating between them is a bit clunky.

We could consider using the [`diffview.nvim`](https://github.com/sindrets/diffview.nvim) plugin and integrating this with [`telescope.nvim`](https://github.com/nvim-telescope/telescope.nvim) so that comparing branches would be like this:

1. Press a mapping like `<leader>gD`
2. Telescope `git_branches` picker lets you pick the base branch.
3. Then you choose the branch to compare with again using `git_branches`.
4. Then `:DiffviewOpen` is called with the head commits of those two branches.

It could be varied so that you choose commits (using `git_commits`) too, instead of just branches.

Finally, it would be great if `diffview.nvim` could show the summary of changed files as it appears in the file tree, with indentation, like this:

```
[M] some_dir/
  [M] file_a.py
  [A] file_b.py
```

Rather than:

```
[M] some_dir/file_a.py
[A] some_dir/file_b.py
```

**Comment by farbodsz** [comment]
Fixed by fde77e295881aac4a7cd55201f80cefc2d177694.

It would still be nice to have the file tree view though so I'll leave this open for now.

**Comment by farbodsz** [comment]
Closing since the PR adding the file tree in `diffview.nvim` has been merged.
